src/frontend/screens/Invites/LeaveProject.tsx

```diff
@@ -0,0 +1,205 @@
+import {Bar as ProgressBar} from 'react-native-progress';
+
+const m = defineMessages({
+  leaveProj: {
+    id: 'screens.Invites.LeaveProject.leaveProj',
+    defaultMessage: 'Leave Project',
+  },
+  deleteConsentWithName: {
+    id: 'screens.Invites.LeaveProject.deleteConsentWithName',
+    defaultMessage:
+      'I understand I will be deleting all data from Project {projectName} from my device.',
+  },
+  deleteConsentWithoutName: {
+    id: 'screens.Invites.LeaveProject.deleteConsentWithoutName',
+    defaultMessage: 'I understand I will be deleting all data from my device.',
+  },
+  removeFromProjWithName: {
+    id: 'screens.Invites.LeaveProject.removeFromProjWithName',
+    defaultMessage:
+      "This will remove all Project {projectName}'s data from your device.",
+  },
+  removeFromProjWithoutName: {
+    id: 'screens.Invites.LeaveProject.removeFromProjWithoutName',
+    defaultMessage: 'This will remove all of the data from your device.',
+  },
+  cancel: {
+    id: 'screens.Invites.LeaveProject.cancel',
+    defaultMessage: 'Cancel',
+  },
+  checkToConfirm: {
+    id: 'screens.Invites.LeaveProject.checkToConfirm',
+    defaultMessage: 'Please check the box to confirm',
+  },
+  leavingProject: {
+    id: 'screens.Invites.LeaveProject.leavingProject',
+    defaultMessage: 'Leaving Project {projectName}',
+  },
+});
+export const LeaveProject = ({
+  route,
+  navigation,
+}: NativeRootNavigationProps<'LeaveProject'>) => {
+  const {formatMessage} = useIntl();
+  const [isChecked, setIsChecked] = useState(false);
+  const [error, setError] = useState(false);
+  const accept = useAcceptInvite();
+  const currentProjectId = useActiveProjectId();
+  const leaveProject = useLeaveProject();
+  const {currentProjectName, inviteId} = route.params;
+  const {data: invite} = useSingleInvite({inviteId});
+  const {setActiveProjectId} = useActiveProjectIdActions();
+
+  useListenToInviteStateUpdate(inviteId);
+
+  function handleLeaveProject() {
+    if (!isChecked) {
+      setError(true);
+      return;
+    }
+
+    // we want to accept first because the invitor will be able to cancel.
+    // this avoids the user leaving a project, and then their invite being cancelled before they were able to join.
+    accept.mutate(
+      {inviteId},
+      {
+        onSuccess: newProjectId => {
+          leaveProject.mutate(
+            {projectId: currentProjectId!},
+            {
+              onSuccess: () => {
+                setActiveProjectId(newProjectId);
+                navigation.replace('InviteSuccessfullyJoined', {
+                  projectName: invite.projectName,
+                });
+              },
+              onError: err => {
+                Sentry.captureException(err);
+                navigation.replace('ErrorBottomSheet');
+              },
+            },
+          );
+        },
+        onError: err => {
+          Sentry.captureException(err);
+          navigation.replace('ErrorBottomSheet');
```

Comment:
out of curiosity, can you explain why this is using `replace`? maybe i'm missing something about how this is set up in the navigation...

Reply:
haven't fully wrapped my head around it, but could this be contributing to the issue I'm seeing in https://github.com/digidem/comapeo-mobile/pull/1080#issuecomment-2776860075 related to the go back button not working?

my initial thought is that this seems fragile and relies on implicit knowledge that isn't entirely clear to me, but i guess i can't really come up with a much stronger suggestion at the moment

Reply:
The alternative is to use `navigate`. That would stack the error screen on top of the invitations screen. But the error has a good chance that it means the invitation is no longer valid. And if that is the case, the user would go back to the invitation screen, and be stuck on it.

The way it is now guarantees that the invitation is only open if it is pending.